packages/api/src/domain/medical/document-query.ts

```diff
@@ -8,12 +8,17 @@ export type Progress = {
   errors?: number;
 };
 
+export type ProgressIntKeys = keyof Omit<Progress, "status">;
+
 export type DocumentQueryProgress = {
   download?: Progress;
   convert?: Progress;
   requestId?: string;
 };
 
+export type ProgressType = keyof Pick<DocumentQueryProgress, "convert" | "download">;
+export const progressTypes = ["convert", "download"] as const;
```

Comment:
NABD, but I'd try to avoid defining the options "convert" and "download" multiple times, could do something like this:

```ts
export const progressTypes = ["convert", "download"] as const;
export type ProgressType = (typeof progressTypes)[number];

export type DocumentQueryProgress = Partial<
  Record<ProgressType, Progress> & {
    requestId: string;
  }
>;
```